sql/core/src/test/resources/sql-tests/inputs/cast.sql

```diff
@@ -42,4 +42,23 @@ SELECT CAST('9223372036854775808' AS long);
 
 DESC FUNCTION boolean;
 DESC FUNCTION EXTENDED boolean;
+
```

Comment:
How about adding more cases?
invalid case: `interval 10 nanoseconds`
valid case: `interval 1 year 2 month 3 week 4 day 5 hour 6 minute -7 seconds 8 millisecond, 9 microsecond`

Reply:
@kiszk fair enough :) I've added test cases for intervals in the wrong order, additions / substractions of intervals, repeating date keywords, embedded spaces etc..

let me know!